src/traces/carpet/axis_attributes.js

```diff
@@ -0,0 +1,445 @@
+    },
+    title: {
+        valType: 'string',
+        role: 'info',
+        description: 'Sets the title of this axis.'
+    },
+    titlefont: extendFlat({}, fontAttrs, {
+        description: [
+            'Sets this axis\' title font.'
+        ].join(' ')
+    }),
+    titleoffset: {
+        valType: 'number',
+        role: 'info',
+        dflt: 10,
+        description: [
+            'An additional amount by which to offset the title from the tick',
+            'labels, given in pixels'
+        ].join(' '),
+    },
+    type: {
+        valType: 'enumerated',
+        // '-' means we haven't yet run autotype or couldn't find any data
+        // it gets turned into linear in gd._fullLayout but not copied back
+        // to gd.data like the others are.
+        values: ['-', 'linear', 'date', 'category'],
+        dflt: '-',
+        role: 'info',
+        description: [
+            'Sets the axis type.',
+            'By default, plotly attempts to determined the axis type',
+            'by looking into the data of the traces that referenced',
+            'the axis in question.'
+        ].join(' ')
+    },
+    autorange: {
+        valType: 'enumerated',
+        values: [true, false, 'reversed'],
+        dflt: true,
+        role: 'style',
+        description: [
+            'Determines whether or not the range of this axis is',
+            'computed in relation to the input data.',
+            'See `rangemode` for more info.',
+            'If `range` is provided, then `autorange` is set to *false*.'
+        ].join(' ')
+    },
+    rangemode: {
+        valType: 'enumerated',
+        values: ['normal', 'tozero', 'nonnegative'],
+        dflt: 'normal',
+        role: 'style',
+        description: [
+            'If *normal*, the range is computed in relation to the extrema',
+            'of the input data.',
+            'If *tozero*`, the range extends to 0,',
+            'regardless of the input data',
+            'If *nonnegative*, the range is non-negative,',
+            'regardless of the input data.'
+        ].join(' ')
+    },
+    range: {
+        valType: 'info_array',
+        role: 'info',
+        items: [
+            {valType: 'any'},
+            {valType: 'any'}
+        ],
+        description: [
+            'Sets the range of this axis.',
+            'If the axis `type` is *log*, then you must take the log of your',
+            'desired range (e.g. to set the range from 1 to 100,',
+            'set the range from 0 to 2).',
+            'If the axis `type` is *date*, it should be date strings,',
+            'like date data, though Date objects and unix milliseconds',
+            'will be accepted and converted to strings.',
+            'If the axis `type` is *category*, it should be numbers,',
+            'using the scale where each category is assigned a serial',
+            'number from zero in the order it appears.'
+        ].join(' ')
+    },
+
+    fixedrange: {
```

Comment:
@rreusser does `fixedrange` do anything on a carpet axis? Can I remove it?

Reply:
🔪